components/base/wheeled/wheeled_base.go

```diff
@@ -347,16 +355,38 @@ func (wb *wheeledBase) differentialDrive(forward, left float64) (float64, float6
 
 // SetVelocity commands the base to move at the input linear and angular velocities.
 func (wb *wheeledBase) SetVelocity(ctx context.Context, linear, angular r3.Vector, extra map[string]interface{}) error {
```

Comment:
Why this `SetVelocity` function needed to change as much as it did is involved. The crux of the problem came from:
* Not registering an operation on the op manager
* Using `runAll` -- which (in this case) calls `GoFor` on the individual motors in separate goroutines.
* Calling `GoFor` with a `0` for the number of revolutions.
  * `GoFor` in that case turns on the motors and returns immediately. For revolutions > 0, GoFor blocks util the motor has gone the requested number of revolutions.

For this code to work "correctly" (i.e: not deadlock), it requires that functions that create operations do *not* reacquire the wheeled base mutex for cleanup.